sdlib/d/gc/region.d

```diff
@@ -609,37 +657,59 @@ unittest trackDirtyBlocks {
 		assert(r.countDirtyBlocksInSubRegion(0, blocks) == dirtyBlocks);
 	}
 
+	// Verify that the currently-dirtiest region has the given attributes.
+	void verifyDirtiestRegion(void* address, uint blocks, uint dirtyBlocks) {
+		Region rr;
+		rr.dirtyBlockCount = uint.max;
+
+		auto r = ra.regionsByDirt.extractBestFit(&rr);
+		assert(r !is null);
+		assert(r.address is address);
+		assert(r.blockCount == blocks);
+		assert(r.dirtyBlockCount == dirtyBlocks);
+		assert(r.countDirtyBlocksInSubRegion(0, blocks) == dirtyBlocks);
+		ra.registerRegionDirt(r);
+	}
+
 	// Initially, there are no dirty blocks.
 	assert(regionAllocator.dirtyBlockCount == 0);
 
 	// Make some dirty regions.
-	freeRun(addresses[0 .. 2]);
-	assert(regionAllocator.dirtyBlockCount == 2);
-	verifyUniqueRegion(addresses[0], 2, 2, 2);
 	freeRun(addresses[4 .. 8]);
-	assert(regionAllocator.dirtyBlockCount == 6);
+	assert(regionAllocator.dirtyBlockCount == 4);
 	verifyUniqueRegion(addresses[4], 4, 4, 4);
+	verifyDirtiestRegion(addresses[4], 4, 4);
 	freeRun(addresses[10 .. 15]);
-	assert(regionAllocator.dirtyBlockCount == 11);
+	assert(regionAllocator.dirtyBlockCount == 9);
 	verifyUniqueRegion(addresses[10], 5, 5, 5);
+	verifyDirtiestRegion(addresses[10], 5, 5);
+	freeRun(addresses[0 .. 2]);
+	assert(regionAllocator.dirtyBlockCount == 11);
+	verifyUniqueRegion(addresses[0], 2, 2, 2);
+	verifyDirtiestRegion(addresses[10], 5, 5);
```

Comment:
Why was the order changed here?

Reply:
I don't get it.

Reply:
It's generally better to add new tests cases for what we want to test, as it prove none of the existing things are broken. Nevertheless, I get it now.

That being said, it seems that every single call to `verifyUniqueRegion` is now followed by a call to `verifyDirtiestRegion`. Then why have two functions?